### PR TITLE
internal/website: rewrite blob how-to guide to use blob godoc examples

### PR DESCRIPTION
**Pull request by zombiezen**
Add Hugo shortcode to read from examples.json.

I rewrote the blob godoc examples to match what was in the how-to guides. The page rendering is slightly different, but no substantive content change.

Updates #1827

### CODE REVIEW

File: blob/azureblob/example_test.go [hunk 1]
```diff
@@ -93,14 +91,21 @@ func Example_sasToken() {
 }
 
 func Example_openBucket() {
+	// This example is used in https://gocloud.dev/howto/blob/open-bucket/#azure
+
+	// import _ "gocloud.dev/blob/azureblob"
```
Comment (eliben) [inline blob/azureblob/example_test.go]: Why is this needed?

Comment (zombiezen) [reply blob/azureblob/example_test.go]: The embedded examples will also show used imports, but it won't show imports not referenced in the example. This is one of the lines gatherexamples processes to add it into the imports shown in the example.

It's useful to show this in documentation because without this blank import, the `blob.OpenBucket` line will return an error (the URL opener won't be linked in).
